### PR TITLE
Allow configuration overrides

### PR DESCRIPTION
**Pull request by PGijsbers**
This PR adds documentation (and the possibility) to allow overrides for the `src/config.toml` configuration and the `.env` configuration. This is useful, because then you can customize your own deployment (also production), without affecting files which are checked into git, which means: 
 - significantly less risk of accidentally checking in your custom configuration (potentially leaking secrets)
 - easier to pull in updates, as previously changed configurations would lead to merge conflicts or otherwise needed to be undone before a pull
 
 I also removed the passing around of environment variables between all docker images, as the relevant variables were already also passed through the `environment` field. Now containers are only aware of the variables they need.
 
 ## How to Test
Change some configurations in `src/config.override.toml` and `override.env` as described in the documentation with some custom options (whatever you are already using, or something new just to test). `up` the containers, poke around, `down` the containers.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -8,7 +8,6 @@ services:
       dockerfile: Dockerfile
     image: aiod_metadata_catalogue
     container_name: apiserver
-    env_file: .env
     environment:
       - KEYCLOAK_CLIENT_SECRET=$KEYCLOAK_CLIENT_SECRET
```
Comment (AdrianAlcolea) [inline docker-compose.yaml:12]: We need to add:
- ES_USER=$ES_USER
- ES_PASSWORD=$ES_PASSWORD
Into the apiserver container environment to connect to the elasticsearch service

File: docker-compose.yaml [hunk 2]
```diff
@@ -34,7 +33,6 @@ services:
     profiles: ["examples"]
     image: aiod_metadata_catalogue
     container_name: fill-db-with-examples
-    env_file: .env
     environment:
       - KEYCLOAK_CLIENT_SECRET=$KEYCLOAK_CLIENT_SECRET
```
Comment (AdrianAlcolea) [inline docker-compose.yaml:40]: We need to add:
- MYSQL_ROOT_PASSWORD=$MYSQL_ROOT_PASSWORD
Into the fill-db-with-examples container to load data into the database

**Review by AdrianAlcolea — COMMENTED** [review]
As we are already talking about that I close the revision here and we can start another one ;p

**Comment by PGijsbers** [comment]
I also had the PYTHON_PATH... I think we should probably just bake that into the docker image, though? Since it's required for the image to function even if no code is mounted... otherwise, things seem to work. All connectors are able to import data now and I can search both directly with the API and through the elastic search endpoints.

**Review by AdrianAlcolea — APPROVED** [review]
Done, everything works for me now =)

**Review by Taniya-Das — APPROVED** [review]
works well for me now.

**Pull request merged**